File: triangle/Triangle.hs
module Triangle where


area :: Double -> Double -> Either String Double
area base height
 | base < 0 && height < 0  = Left "base and height cannot be negative"
 | base < 0  = Left "base cannot be negative"
 | height < 0  = Left "height cannot be negative"
 | otherwise = Right $ height * base / 2

File: src/main/java/za/co/anna/monde/triangle/Triangle.java
package za.co.anna.monde.triangle;

class Triangle {

    public static double area( double base, double height )
            throws IllegalArgumentException{
        if ( base < 0 && height < 0 )
            throw new IllegalArgumentException(
                    "base and height cannot be negative"
            );

        if ( base < 0 )
            throw new IllegalArgumentException(
                    "base cannot be negative"
            );

        if ( height < 0 )
            throw new IllegalArgumentException(
                    "height cannot be negative"
            );

        return height * base / 2;
    }
}
